### PR TITLE
Refactor UserSignIn functionality

### PR DESCRIPTION
**Pull request by gimochi**
## Related Issues
#80 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (one that would cause existing functionality to change)

## Testing
- [ ] Tested in some way, if yes, the description follows.

### CODE REVIEW

File: src/dao/User/UserSelectDAO.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
 import db from '../../database';
+import { UserStatus } from "../../models/UserStatus";
 
 export const selectUserByEmail = (email: string) => {
   return db('User').where({
     email: email
   })
+  .whereNot({
```
Comment (enginehenryed) [inline src/dao/User/UserSelectDAO.ts]:
```javascript
func().func1()
  .func2()
  .func3()
```

File: src/models/ErrorType/UserError.ts [hunk 2]
```diff
@@ -28,5 +28,10 @@ export default {
     type: 'EMAIL_EXIST',
     code: 401005,
     msg: 'email exit'
+  },
+  'PENDING_USER': {
```
Comment (enginehenryed) [inline src/models/ErrorType/UserError.ts]: Should better start with _Noun_, like in `USER_PENDING`

File: src/models/ErrorType/UserError.ts [hunk 3]
```diff
@@ -28,5 +28,10 @@ export default {
     type: 'EMAIL_EXIST',
     code: 401005,
     msg: 'email exit'
```
Comment (enginehenryed) [inline src/models/ErrorType/UserError.ts]: typo

File: src/models/ErrorType/UserError.ts [hunk 4]
```diff
@@ -28,5 +28,10 @@ export default {
     type: 'EMAIL_EXIST',
```
Comment (enginehenryed) [inline src/models/ErrorType/UserError.ts]: Can better be e.g. `USER_EMAIL_ALREADY_DEFINED`

File: src/routes/v1/userRoute.ts [hunk 5]
```diff
@@ -47,7 +48,10 @@ function userRoute(router) {
      * 로그인
      */
     .post((request: Request, response: Response) => {
-      //TODO
+      const req: UserRequest.SignIn = request.body;
+      const payload = UserSignInController.signInUser(req);
+      console.log(1, 'hi')
```
Comment (enginehenryed) [inline src/routes/v1/userRoute.ts]: Remove

**Review by eldenpark — CHANGES_REQUESTED** [review]
Very good

File: src/dao/Definition/DefinitionSelectDAO.ts [hunk 6]
```diff
@@ -7,6 +7,7 @@ export const getDefinitions = () => {
 
 export const getDefinitionsByTermId = (termId: number) => {
   return db('Definition').where({
-    term_id: termId
-  }).select()
+      term_id: termId
+    })
+    .select()
```
Comment (enginehenryed) [inline src/dao/Definition/DefinitionSelectDAO.ts]: `;`

**Pull request merged**